source/functional-concepts.md

```diff
@@ -0,0 +1,150 @@
+# Functional Programming in Swift: A Primer
+
+Swift is a hybrid language that allows both objective-oriented and functional code to co-exist.
+
+### First-class functions
+
+First-class functions are *values* that can either take other functions as arguments or return them as results. Also refered to as *higher order functions*.
+
+#### Returning Functions
+
+~~~swift
+typealias Prefixer = String -> String
+
+func prefixerFactory(prefix: String) -> Prefixer {
+// func prefix(prefix: String) -> (String -> String)
+
+  func aPrefixer(word: String) -> String {
+    return prefix + word
+  }
+
+  return aPrefixer
+}
+
+let repeat = prefixerFactory("re-")
+repeat("do")
+~~~
+
+*Note*: You can also return an unnamed function, i.e. a closure.
```

Comment:
Are those official Swift technical terms? "An unnamed function" is usually called "an anonymous function", "a lambda", or "a function object" in, like, every other programming language and in computer science, so I'd be hesitant to call it something different here unless the Swift docs specifically do so.

And I know the Swift docs call lambdas closures, but they are not the same thing and repeating their mistake makes me sad.